### PR TITLE
Extend MAU Types

### PR DESCRIPTION
**Pull request by axkar**
Extended MAU Type Support with 2.5G, 5G and 10G (Copper and Fiber)

### CODE REVIEW

File: src/ports/linux/pnal.c [hunk 1]
```diff
@@ -307,11 +307,17 @@ static pnal_eth_mau_t calculate_mau_type (
          switch (speed)
          {
          case SPEED_10:
-            return PNAL_ETH_MAU_COPPER_10BaseT;
+            return PNAL_ETH_MAU_COPPER_10BaseTX_FULL_DUPLEX;
```
Comment (troglobit) [inline src/ports/linux/pnal.c]: Had a brief review with @wkz and this was the only thing he got a knee-jerk reaction to.  Turns out I added this misnomer in 166f71df.  Would be great if you could add a commit to this branch that corrects my change.  

I.e., rename all `*10BaseTX*` to `*10BaseT*` since -TX only arrived with 100 Mbps.  (ref: https://en.wikipedia.org/wiki/Ethernet_over_twisted_pair)

File: src/device/pf_port.c [hunk 2]
```diff
@@ -225,15 +225,23 @@ pf_mediatype_values_t pf_port_get_media_type (pnal_eth_mau_t mau_type)
...
+   case PNAL_ETH_MAU_COPPER_5000BaseT:
+   case PNAL_ETH_MAU_COPPER_2500BaseT:
       return PF_PD_MEDIATYPE_COPPER;
    case PNAL_ETH_MAU_FIBER_100BaseFX_HALF_DUPLEX:
    case PNAL_ETH_MAU_FIBER_100BaseFX_FULL_DUPLEX:
    case PNAL_ETH_MAU_FIBER_1000BaseX_HALF_DUPLEX:
    case PNAL_ETH_MAU_FIBER_1000BaseX_FULL_DUPLEX:
+   case PNAL_ETH_MAU_FIBER_10000BaseR:
+   case PNAL_ETH_MAU_FIBER_2500BaseX:
+   case PNAL_ETH_MAU_FIBER_5000BaseR:
```
Comment (troglobit) [inline src/device/pf_port.c]: This is a *super minor* detail, but maybe we should order all these lists according to size/speed?  Like you did below.

File: src/device/pf_port.c [hunk 3]
```diff
@@ -225,15 +225,23 @@ pf_mediatype_values_t pf_port_get_media_type (pnal_eth_mau_t mau_type)
    case PNAL_ETH_MAU_RADIO:
       return PF_PD_MEDIATYPE_RADIO;
    case PNAL_ETH_MAU_COPPER_10BaseT:
+   case PNAL_ETH_MAU_COPPER_10BaseTX_HALF_DUPLEX:
+   case PNAL_ETH_MAU_COPPER_10BaseTX_FULL_DUPLEX:
```
Comment (troglobit) [inline src/device/pf_port.c]: See comment below, suggestion to rename my badly named enums 10BaseTX -> 10BaseT.

**Review by troglobit — CHANGES_REQUESTED** [review]
Overall very clean and readable change.  All I was missing, on the whole, was a reference to the standard, since I don't know the MauType numbers, so I have no way of checking them for accuracy.

Also, again minor detail, but a summary of changes in the PR description. No need to repeat a commit message, but more of an overall summary.  Example: your current title is a bit long and could've been the PR description while the title could've been simply "Extend MAU Types"

**Comment by axkar** [comment]
IANAifMauTypeListBits :: "[IEEE802.3], Section 30.5.1.1.2"
https://www.iana.org/assignments/ianamau-mib/ianamau-mib

File: src/device/pf_port.c (hunk 2 above)
Comment (axkar) [reply src/device/pf_port.c]: Earlier they were sorted according to the bit positions, but I agree, it is more readable when they sorted/filtered in this order:
1. Speed
2. Duplex
3. Connection-type: Copper/Fiber

Comment (troglobit) [reply src/device/pf_port.c]: Great, I agree!

File: src/device/pf_port.c [hunk 4]
```diff
@@ -225,23 +225,23 @@ pf_mediatype_values_t pf_port_get_media_type (pnal_eth_mau_t mau_type)
    case PNAL_ETH_MAU_RADIO:
       return PF_PD_MEDIATYPE_RADIO;
    case PNAL_ETH_MAU_COPPER_10BaseT:
-   case PNAL_ETH_MAU_COPPER_10BaseTX_HALF_DUPLEX:
-   case PNAL_ETH_MAU_COPPER_10BaseTX_FULL_DUPLEX:
+   case PNAL_ETH_MAU_COPPER_10BaseT_HALF_DUPLEX:
+   case PNAL_ETH_MAU_COPPER_10BaseT_FULL_DUPLEX:
```
Comment (troglobit) [inline src/device/pf_port.c]: This rename part of the commit should be in a separate commit from the below reorder.

File: src/device/pf_port.c [hunk 5]
```diff
@@ -225,23 +225,23 @@ pf_mediatype_values_t pf_port_get_media_type (pnal_eth_mau_t mau_type)
...
    case PNAL_ETH_MAU_COPPER_100BaseTX_HALF_DUPLEX:
    case PNAL_ETH_MAU_COPPER_100BaseTX_FULL_DUPLEX:
    case PNAL_ETH_MAU_COPPER_1000BaseT_HALF_DUPLEX:
    case PNAL_ETH_MAU_COPPER_1000BaseT_FULL_DUPLEX:
-   case PNAL_ETH_MAU_COPPER_10000BaseT:
-   case PNAL_ETH_MAU_COPPER_5000BaseT:
    case PNAL_ETH_MAU_COPPER_2500BaseT:
+   case PNAL_ETH_MAU_COPPER_5000BaseT:
+   case PNAL_ETH_MAU_COPPER_10000BaseT:
```
Comment (troglobit) [inline src/device/pf_port.c]: This reorder could be squashed with the previous commit.

File: src/device/pf_port.c [hunk 6]
```diff
@@ -225,23 +225,23 @@ pf_mediatype_values_t pf_port_get_media_type (pnal_eth_mau_t mau_type)
...
       return PF_PD_MEDIATYPE_COPPER;
    case PNAL_ETH_MAU_FIBER_100BaseFX_HALF_DUPLEX:
    case PNAL_ETH_MAU_FIBER_100BaseFX_FULL_DUPLEX:
    case PNAL_ETH_MAU_FIBER_1000BaseX_HALF_DUPLEX:
    case PNAL_ETH_MAU_FIBER_1000BaseX_FULL_DUPLEX:
-   case PNAL_ETH_MAU_FIBER_10000BaseR:
    case PNAL_ETH_MAU_FIBER_2500BaseX:
    case PNAL_ETH_MAU_FIBER_5000BaseR:
+   case PNAL_ETH_MAU_FIBER_10000BaseR:
```
Comment (troglobit) [inline src/device/pf_port.c]: same here (squash with prev.)

File: src/pnal.h [hunk 7]
```diff
@@ -84,9 +84,9 @@ typedef enum pnal_eth_mau
 {
    PNAL_ETH_MAU_UNKNOWN = 0x00, /* When link is down */
    PNAL_ETH_MAU_RADIO = 0x00,   /* When link is up */
-   PNAL_ETH_MAU_COPPER_10BaseT = 0x05,
-   PNAL_ETH_MAU_COPPER_10BaseTX_HALF_DUPLEX = 0x0A,
-   PNAL_ETH_MAU_COPPER_10BaseTX_FULL_DUPLEX = 0x0B,
+   PNAL_ETH_MAU_COPPER_10BaseT = 0x05, /* duplex mode unknown */
+   PNAL_ETH_MAU_COPPER_10BaseT_HALF_DUPLEX = 0x0A,
+   PNAL_ETH_MAU_COPPER_10BaseT_FULL_DUPLEX = 0x0B,
```
Comment (troglobit) [inline src/pnal.h:89]: same, re: separate commit.

File: src/pnal.h [hunk 8]
```diff
@@ -95,12 +95,12 @@ typedef enum pnal_eth_mau
...
    PNAL_ETH_MAU_FIBER_1000BaseX_FULL_DUPLEX = 0x16,
-   PNAL_ETH_MAU_FIBER_10000BaseR = 0x21,
-   PNAL_ETH_MAU_COPPER_10000BaseT = 0x36,
    PNAL_ETH_MAU_COPPER_2500BaseT = 0x67,
-   PNAL_ETH_MAU_COPPER_5000BaseT = 0x68,
    PNAL_ETH_MAU_FIBER_2500BaseX = 0x6E,
+   PNAL_ETH_MAU_COPPER_5000BaseT = 0x68,
    PNAL_ETH_MAU_FIBER_5000BaseR = 0x70,
+   PNAL_ETH_MAU_COPPER_10000BaseT = 0x36,
+   PNAL_ETH_MAU_FIBER_10000BaseR = 0x21,
```
Comment (troglobit) [inline src/pnal.h]: same as previously

File: src/ports/linux/pnal.c [hunk 9]
```diff
@@ -307,7 +307,7 @@ static pnal_eth_mau_t calculate_mau_type (
          switch (speed)
          {
          case SPEED_10:
-            return PNAL_ETH_MAU_COPPER_10BaseTX_FULL_DUPLEX;
```
Comment (troglobit) [inline src/ports/linux/pnal.c]: same as previously mentioned

File: src/ports/linux/pnal.c [hunk 10]
```diff
@@ -327,7 +327,7 @@ static pnal_eth_mau_t calculate_mau_type (
          switch (speed)
          {
          case SPEED_10:
-            return PNAL_ETH_MAU_COPPER_10BaseTX_HALF_DUPLEX;
+            return PNAL_ETH_MAU_COPPER_10BaseT_HALF_DUPLEX;
```
Comment (troglobit) [inline src/ports/linux/pnal.c]: same as previously mentioned

**Review by troglobit — CHANGES_REQUESTED** [review]
Technically correct, just some minor formality about where to put changes vs commits.

The link to the iana mau types can be included in the original/first commit message when you rebase + squash this.

**Comment by troglobit** [comment]
> IANAifMauTypeListBits :: "[IEEE802.3], Section 30.5.1.1.2" https://www.iana.org/assignments/ianamau-mib/ianamau-mib

This would be great to have in the commit (the PR text + conversation is not part of the commit history of the project.).

**Review by troglobit — APPROVED** [review]

**Comment by troglobit** [comment]
Nice work on this!
